server/bundler/loader.js

```diff
@@ -102,26 +130,10 @@ function splitTemplate( path, section ) {
 		'		return;',
 		'	}',
 		'	context.store.dispatch( { type: "SECTION_SET", isLoading: true } );',
-		'	require.ensure([], function( require ) {',
-		'		context.store.dispatch( { type: "SECTION_SET", isLoading: false } );',
-		'		controller.setSection( ' + sectionString + ' )( context );',
-		'		if ( ! _loadedSections[ ' + moduleString + ' ] ) {',
-		'			require( ' + moduleString + ' )( controller.clientRouter );',
-		'			_loadedSections[ ' + moduleString + ' ] = true;',
-		'		}',
-		'		context.store.dispatch( activateNextLayoutFocus() );',
-		'		next();',
-		'	}, function onError( error ) {',
-		'		if ( ! LoadingError.isRetry() ) {',
-		'			LoadingError.retry( ' + sectionNameString + ' );',
-		'		} else {',
-		'			console.error(error);',
-		'			context.store.dispatch( { type: "SECTION_SET", isLoading: false } );',
-		'			LoadingError.show( ' + sectionNameString + ' );',
-		'		}',
-		'		return;',
-		'	},',
-		sectionNameString + ' );',
+		'   const moduleId = require.resolve( ' + moduleString + ');',
+		'   const callback = requireCallback( context, next, ' + sectionString + ', ' + moduleString + ', moduleId );',
+		'   const onError = onErrorCallback( context, ' + sectionNameString + ' );',
+		'   require.ensure([], callback, onError, ' + sectionNameString + ' );',
```

Comment:
It seems code splitting isn't working anymore with this change.
That's because webpack needs to determine the dependencies (each `require` calls) of each chunk statically (from the code generated). So we must have this kind of syntax:
```js
const moduleId = '...';
require.ensure([], function(require) {
    require( moduleId )();
});
```
We can however refactor everything else, before and after the `require` call.

Reply:
I see... you mean after and before the `require.ensure` call, right?

Reply:
Well both, you can group everything else in reusable functions as long as you keep `require.ensure` containing the `require` calls.